apps/air-discount-scheme/web/infra/web.ts

```diff
@@ -7,12 +7,20 @@ export const serviceSetup = (services: {
     .namespace('air-discount-scheme')
     .env({
       API_URL: ref((h) => `http://${h.svc(services.adsApi)}`),
+      NEXT_PUBLIC_NEXTAUTH_URL: {
+        dev: 'https://loftbru.dev01.devland.is',
+        staging: 'https://loftbru.staging01.devland.is',
+        prod: 'https://loftbru.island.is',
+      },
+      NEXT_PUBLIC_IDENTITY_SERVER_DOMAIN: {
+        dev: 'identity-server.dev01.devland.is',
+        staging: 'identity-server.staging01.devland.is',
+        prod: 'innskra.island.is',
+      },
```

Comment:
@pshomov @Snojo This is not working, on build it throws an error. What is the best practice when having infra values available for web production builds ?

Reply:
I think it depends on whether the next backend or client code is referencing this.

If the client code is referencing it you will probably need to use the nextjs publicruntimeconfig

Reply:
I suppose it wouldnt work for local environment. This is how it would be running locally:
`{"loc":"http://localhost:4200","dev":"https://loftbru.dev01.devland.is","staging":"https://loftbru.staging01.devland.is","prod":"https://loftbru.island.is"}`
`[next-auth][error][signin_oauth_error] 
https://next-auth.js.org/errors#signin_oauth_error Invalid URL: https://[object Object]/connect/authorize?response_type=code`

Reply:
Okay if I understand the Nextauth code correctly it is not being run in the browser so you don't need the publicruntimeconfig.

For anything local you would set the variable in the devConfig object in the environment.ts file.

For dev, staging and prod you would set the process.env.BLA in the prodConfig object in the environment.ts file and set the `BLA: { dev: 'stuff', staging: 'stuff', prod: 'stuff }` inside the infra/web.ts file

Reply:
But if you are unsure that this works after it's deployed, you can set the `deploy feature` label on the PR and you can try this out without merging the PR.